### PR TITLE
Run shellcheck on all bash code

### PR DESCRIPTION
**Pull request by lack**
Signed-off-by: Jim Ramsay <jramsay@redhat.com>

### CODE REVIEW

**Comment by lack** [comment]
@Miciah: This should address all your comments on the `kubensenter` script from https://github.com/openshift/kubernetes/pull/1327

Once this is merged I'll release v1.1.2 from this repo and import it there.

File: test/library-integration.bats [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
   bats_require_minimum_version 1.5.0
 
 DIR="$( cd "$( dirname "$BATS_TEST_FILENAME" )" >/dev/null 2>&1 && pwd )"
-. $DIR/utils.sh
+. "$DIR/utils.sh"
 
 function setup_file() {
     setup_namespaces
+
+    # Ensure the test binaries (build alongside this file) are in the path
```
Comment (sakhoury) [inline test/library-integration.bats]: Update comment: `...(build alongside ...)..` to `...(built alongside ...)..`

File: test/utils.sh [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
+    MOUNT_NAMESPACE=${TESTDIR}/mnt
+    touch "$MOUNT_NAMESPACE"
+    unshare --mount="$MOUNT_NAMESPACE" --propagation slave mount --make-rshared /
+    NEW_NS=$(nsenter -m"$MOUNT_NAMESPACE" readlink /proc/self/ns/mnt)
+    OLD_NS=$(readlink /proc/self/ns/mnt)
 
-    export ALT_NAMESPACE=${TESTDIR}/net
-    touch $ALT_NAMESPACE
-    unshare --net=$ALT_NAMESPACE true
+    # Set up a not-a-mount namespace for negative tests
```
Comment (sakhoury) [inline test/utils.sh]: Update comment: `... not-a-mount ...` to `... non-mount ...`

File: utils/kubensenter/kubensenter [hunk 3]
```diff
@@ -1,43 +1,54 @@
...
-    echo "user-specified command to nsenter(1). If kubens.service has not set up"
-    echo "the mount namespace, the user-specified command is still passed to"
-    echo "nsenter(1) but no namespace is entered."
+    echo 'Autodetect whether the `kubens.service` has pinned a mount namespace in a'
+    echo 'well-known location, and if so, join it by passing it and the user-specified'
+    echo 'command to nsenter(1). If `kubens.service` has not set up the mount namespace,'
+    echo 'the user-specified command is still executed by nsenter(1) but no namespace is'
+    echo 'entered.'
+    echo
+    echo 'If $KUBENSMNT is set in the environment, skip autodetectipn and attempt to join'
```
Comment (sakhoury) [inline utils/kubensenter/kubensenter]: `autodetectipn` to `autodetection`

**Review by sakhoury — CHANGES_REQUESTED** [review]

**Review by sakhoury — APPROVED** [review]
\lgtm

**Pull request merged**